libs/judicial-system/message/src/lib/message.service.ts

```diff
@@ -189,22 +210,29 @@ export class MessageService {
       .catch((err) => {
         this.connectToQueue()
 
-        throw err
+        if (retry) {
+          throw err
+        }
+
+        // Retry once
+        return this.sendMessageToQueue(message, true)
```

Comment:
We do not want to wait for a connection if we are retrying because then we want to throw an exception. When not retrying, we do not need to wait for `this.connectToQueue()` because the call to `this.sendMessageToQueue` will end up waiting for the promise created by `this.connectToQueue()`.